src/IPStack/Packet/IPPacket.swift

```diff
@@ -271,17 +271,19 @@ open class IPPacket {
         setPayloadWithUInt16(offset, at: 6)
         setPayloadWithUInt8(TTL, at: 8)
         setPayloadWithUInt8(transportProtocol.rawValue, at: 9)
-        // clear checksum bytes
-        resetPayloadAt(10, length: 2)
+
         setPayloadWithUInt32(sourceAddress.UInt32InNetworkOrder!, at: 12, swap: false)
         setPayloadWithUInt32(destinationAddress.UInt32InNetworkOrder!, at: 16, swap: false)
 
+        // set IP checksum
+        resetPayloadAt(10, length: 2)
+        setPayloadWithUInt16(Checksum.computeChecksum(packetData, from: 0, to: Int(headerLength)), at: 10, swap: false)
+        
```

Comment:
I think it is fine since I'm probably not going to maintain this, this change won't be a problem.
But since there are still memory leaks, I do not think it is useable no matter this is changed or not.